src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -1145,15 +1155,34 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
     };
 
     renderActivitySidebarFilter = () => {
-        const { features } = this.props;
+        const { features, hasReplies, hasTasks } = this.props;
         const { feedItemsStatusFilter } = this.state;
-        const shouldShowActivityFeedFilter = isFeatureEnabled(features, 'activityFeed.filter.enabled');
+        const activityFeedFilterEnabled = isFeatureEnabled(features, 'activityFeed.filter.enabled');
+        const newThreadedRepliesEnabled = isFeatureEnabled(features, 'activityFeed.newThreadedReplies.enabled');
+        const shouldShowActivityFeedFilter =
+            activityFeedFilterEnabled || (newThreadedRepliesEnabled && (hasReplies || hasTasks));
 
         if (!shouldShowActivityFeedFilter) {
             return null;
         }
+
+        const activityFilterOptions: ActivityFilterOption[] = [ACTIVITY_FILTER_OPTION_ALL];
+        if (newThreadedRepliesEnabled) {
+            // Determine which filter options to show based on what activity types are available in current context
+            if (hasReplies) {
+                activityFilterOptions.push(ACTIVITY_FILTER_OPTION_OPEN, ACTIVITY_FILTER_OPTION_RESOLVED);
+            }
+            if (hasTasks) {
+                activityFilterOptions.push(ACTIVITY_FILTER_OPTION_TASKS);
+            }
+        } else if (activityFeedFilterEnabled) {
```

Comment:
Not sure if these conditionals are correct.
I think regardless of `newThreadedRepliesEnabled`. `activityFeedFilterEnabled` needs to be enabled in order to show the filter options